packages/core/admin/admin/src/pages/AuthPage/components/Register/index.js

```diff
@@ -33,26 +37,70 @@ const A = styled.a`
   color: ${({ theme }) => theme.colors.primary600};
 `;
 
-const Register = ({ fieldsToDisable, noSignin, onSubmit, modifiedData, schema }) => {
+const Register = ({ fieldsToDisable, noSignin, onSubmit, schema }) => {
+  const toggleNotification = useNotification();
+  const { push } = useHistory();
   const [passwordShown, setPasswordShown] = useState(false);
   const [confirmPasswordShown, setConfirmPasswordShown] = useState(false);
+  const [userInfo, setUserInfo] = useState({});
   const { formatMessage } = useIntl();
+  const query = useQuery();
+  const registrationToken = query.get('registrationToken');
+
+  useEffect(() => {
+    if (registrationToken) {
+      const getData = async () => {
+        try {
+          const {
+            data: { data },
+          } = await axios.get(
+            `${strapi.backendURL}/admin/registration-info?registrationToken=${registrationToken}`
+          );
+
+          if (data) {
+            setUserInfo(data);
+          }
+        } catch (err) {
+          const errorMessage = get(err, ['response', 'data', 'message'], 'An error occurred');
+
+          toggleNotification({
+            type: 'warning',
+            message: errorMessage,
+          });
+
+          // Redirect to the oops page in case of an invalid token
+          // @alexandrebodin @JAB I am not sure it is the wanted behavior
```

Comment:
Yes it is!

Reply:
That was an old comment from you 😄 . Are we sure about that?

Reply:
Yes